### PR TITLE
[Python] Improve Python consistency: Use function_name(…) throughout (PEP8)

### PR DESCRIPTION
**Pull request by practicalswift**
Background: See discussion with @modocache and @nadavrot in #1369. This is a follow-up to #1374.

The majority of Python function names in the repo are in lowercase with words separated by underscores (`snake_case_function`). This is also the form recommended by the [PEP8 function naming recommendations](https://www.python.org/dev/peps/pep-0008/#function-names) (Python Software Foundation).

This PR improves consistency by converting the remaning cases of `camelCaseFunction` to the PEP-8 form (`snake_case_function`) used by convention in the repo.

@modocache - do you have time to review? That would be really appreciated :-)

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
Awesome! What's left before we can activate the `.pep8` rule for method/variable names?

File: tools/SourceKit/bindings/python/sourcekitd/__init__.py [hunk 1]
```diff
@@ -21,4 +21,4 @@
     Bindings for the sourcekitd framework.
 """
 
-__all__ = ['capi', 'request']
+__all__ = ('capi', 'request')
```
Comment (modocache) [inline tools/SourceKit/bindings/python/sourcekitd/__init__.py]: All of these transformations to `__all__` are probably incorrect, no? As far as I know, `__all__` is a **list**, not a tuple.

Comment (modocache) [inline tools/SourceKit/bindings/python/sourcekitd/__init__.py]: Did your search and replace misfire?

**Comment by modocache** [comment]
This looks great, aside from the changes to `__all__`--the documentation indicates that it should be a list: https://docs.python.org/2/tutorial/modules.html#importing-from-a-package. I have a feeling the changes to `__all__` will cause CI to fail here.

**Comment by practicalswift** [comment]
@modocache Thanks for reviewing! :-)

`pydocstyle`/`pep257` requires `__all__` to be a tuple. See discussion here: https://github.com/PyCQA/pydocstyle/issues/62

When thinking of it the `__all__` change has nothing to do with function naming, so I've reverted that change in 23abb10e8584e4f88d6008dd40c7aade6b8a4da0 which is now part of this PR.

Thanks for noticing!

**Comment by practicalswift** [comment]
@gribozavr Merge conflicts have now been resolved :-)

**Comment by practicalswift** [comment]
@gribozavr + @nadavrot Another round of newly introduced merge conflicts resolved :-)

Let me know if any changes are needed for this PR to be merged.

**Comment by practicalswift** [comment]
@modocache Regarding your question about the `.pep8` rule for method/variable names - do you mean the rules in `pep8-naming`? :-)

**Comment by gottesmm** [comment]
@swift-ci Please test

**Comment by practicalswift** [comment]
@gottesmm Both builds passed. Are there any further changes needed for this PR to be merged? :-)

**Comment by gottesmm** [comment]
@practicalswift Out of curiosity, how did you generate the changes? Were you using something like rope?

**Comment by gottesmm** [comment]
@practicalswift The reason why I am asking is I want to know if I should lint the code while reviewing since some of this code is not invoked by the normal build.

**Comment by practicalswift** [comment]
@gottesmm All changes were made manually using `replace-string` in Emacs :-)   However, I used `flake8` to identify the issues.

**Comment by practicalswift** [comment]
Another newly introduced merge conflict now resolved :-)

**Comment by practicalswift** [comment]
@gottesmm This oneliner can be used to list of the renamed functions:

```
$ git log -u -n 1 | egrep '^-.*def ' | cut -b2- | sed 's/^ *//g' | cut -f1 -d'(' | awk '{ print $2 }' | sort | uniq
accumulateCode
addFunction
addLine
addPass
appendText
bodyLines
codeStartsWithDedentKeyword
compareFunctionSizes
compareScores
compareSizes
compareSizesOfFile
compileOpaqueCFile
compileSource
compileSources
computeItersNumber
executeTemplate
formatChildren
get_BMP_data_offset
get_BMP_first_level_index
get_grapheme_cluster_break_tests_as_UTF8
getLineStarts
getScores
getSuccs
_int_list_to_LE_bytes
_int_to_LE_bytes
isMaxScore
listFunctionSizes
nextToken
numeric_type_names_Macintosh_only
parseArguments
parseBenchmarkOutput
parseGenericOperator
parseInt
parseProtocol
parseTemplate
posToLine
Print
printBestScores
printCommand
Process
processSource
processSources
readSizes
reportResults
runBench
runBenchmarks
runCommand
splitGybLines
splitLines
stripTrailingNL
submit_to_LNT
tokenGenerator
tokenizePythonToUnmatchedCloseCurly
tokenizeTemplate
tokenPosToIndex
```

**Comment by practicalswift** [comment]
And this to search for occurrences of the old versions of the now renamed functions:

```
$ for OLD_NAME in $(git log -u -n 1 | egrep '^-.*def ' | cut -b2- | sed 's/^ *//g' | cut -f1 -d'(' | awk '{ print $2 }' | sort | uniq); do echo "${OLD_NAME}: "; git grep -e "${OLD_NAME}[^a-zA-Z]" | egrep -v '\.(cpp|h|swift|sil|md|txt|json|mm|td|def):'; done
accumulateCode:
addFunction:
addLine:
addPass:
appendText:
bodyLines:
codeStartsWithDedentKeyword:
compareFunctionSizes:
compareScores:
compareSizes:
compareSizesOfFile:
compileOpaqueCFile:
compileSource:
compileSources:
computeItersNumber:
executeTemplate:
formatChildren:
get_BMP_data_offset:
get_BMP_first_level_index:
get_grapheme_cluster_break_tests_as_UTF8:
getLineStarts:
getScores:
getSuccs:
_int_list_to_LE_bytes:
_int_to_LE_bytes:
isMaxScore:
listFunctionSizes:
nextToken:
numeric_type_names_Macintosh_only:
parseArguments:
parseBenchmarkOutput:
parseGenericOperator:
parseInt:
parseProtocol:
parseTemplate:
posToLine:
Print:
stdlib/public/core/Arrays.swift.gyb:      debugPrint(item, terminator: "", toStream: &result)
stdlib/public/core/HashedCollections.swift.gyb:      debugPrint(member, terminator: "", toStream: &result)
stdlib/public/core/HashedCollections.swift.gyb:      debugPrint(k, terminator: "", toStream: &result)
stdlib/public/core/HashedCollections.swift.gyb:      debugPrint(v, terminator: "", toStream: &result)
utils/build-script-impl:# Print instructions for using this script to stdout
utils/error_enum_to_cases.pl:# Print properties for the ranges.
utils/find-unused-diagnostics.sh:# Print all diags that occur in the .td files but not in the source.
utils/swift-api-dump.py:    parser.add_argument('-v', '--verbose', action='store_true', help='Print extra information.')
utils/swift-api-dump.py:# Print out the command we're about to execute
utils/swift_build_support/swift_build_support/debug.py:# swift_build_support/debug.py - Print information on the build -*- python -*-
utils/swift_build_support/swift_build_support/debug.py:	Print the host machine's `xcodebuild` version, as well as version
printBestScores:
printCommand:
Process:
benchmark/scripts/Benchmark_Driver:            print "Process output:\n", e.output
benchmark/scripts/Benchmark_RuntimeLeaksRunner.in:            print("Child Process Failed! (%s,%s)" % (data['path'], data['test_name']))
cmake/modules/AddSwift.cmake:# Process the sources within the given variable, pulling out any Swift
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:  print("$ \(Process.arguments[0]) " +
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:func _childProcess() {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:struct _ParentProcess {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:  internal var _runTestsInProcess: Bool
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:  init(runTestsInProcess: Bool, args: [String], filter: String?) {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    self._runTestsInProcess = runTestsInProcess
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:        if _runTestsInProcess {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:  let _isChildProcess: Bool =
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    Process.arguments.contains("--stdlib-unittest-run-child")
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:  if _isChildProcess {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    _childProcess()
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    var runTestsInProcess: Bool = false
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    while i < Process.arguments.count {
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:      let arg = Process.arguments[i]
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:        runTestsInProcess = true
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:        filter = Process.arguments[i + 1]
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:      args.append(Process.arguments[i])
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:    var parent = _ParentProcess(
stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb:      runTestsInProcess: runTestsInProcess, args: args, filter: filter)
utils/profdata_merge/process.py:class ProfdataMergerProcess(Process):
utils/profdata_merge/process.py:        super(ProfdataMergerProcess, self).__init__()
utils/profdata_merge/runner.py:    processes = [ProfdataMergerProcess(config, file_queue) for _ in range(10)]
utils/profdata_merge/runner.py:    merge_final = ProfdataMergerProcess(config, file_queue)
utils/swift-bench.py:  if Process.arguments.count > 1 {
utils/swift-bench.py:    N = Process.arguments[1].toInt()!
utils/swift-bench.py:  if Process.arguments.count <= 2 || Process.arguments[2] == name {
processSource:
processSources:
readSizes:
reportResults:
runBench:
runBenchmarks:
runCommand:
splitGybLines:
splitLines:
stripTrailingNL:
submit_to_LNT:
tokenGenerator:
tokenizePythonToUnmatchedCloseCurly:
tokenizeTemplate:
tokenPosToIndex:
```

**Comment by gottesmm** [comment]
Ok. I am going to just run it through pylint and then I will merge assuming nothing comes up.

**Comment by practicalswift** [comment]
@gottesmm Excellent! :+1:  Thanks!

**Comment by gottesmm** [comment]
pylint -E was not clean, but none of it was due to this commit. This commit LGTM.

**Pull request merged**